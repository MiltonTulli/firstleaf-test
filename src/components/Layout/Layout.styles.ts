import { makeStyles } from '@material-ui/core/styles';
import { Theme } from '@material-ui/core';

export default makeStyles<Theme>(theme => ({
  root: {
    display: 'flex',
    height: '100vh'
  },
  appBar: {
    backgroundColor: theme.palette.primary.main
  },
  container: {
    [theme.breakpoints.down('xs')]: {
      justifyContent: 'center'
    }
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  hide: {
    display: 'none'
  },
  content: {
    paddingTop: 64 + theme.spacing(1),
    padding: theme.spacing(1),
    width: '100%',
    [theme.breakpoints.down('xs')]: {
      paddingTop: 56
    }
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen
    }),
    marginLeft: '0 !important'
  },
  link: {
    color: 'inherit',
    textDecoration: 'none'
  }
}));
