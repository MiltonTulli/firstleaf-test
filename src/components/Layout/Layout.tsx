import React from 'react';
import classNames from 'classnames';
import _isEqual from 'lodash/isEqual';
import { Link, useRouteMatch } from 'react-router-dom';
import { AppBar, Toolbar, Typography, Grid } from '@material-ui/core';
import { SearchInput, useDimensions } from '..';
import useStyles from './Layout.styles';
import R from '../../utils/routes';

interface ILayoutProps {
  children: React.ReactNode;
  handleSearch: (country: string) => void;
  filter?: boolean;
}

export const Layout: React.FC<ILayoutProps> = (props: ILayoutProps) => {
  const { children, handleSearch, filter = false } = props;
  const { isMobile } = useDimensions();
  const classes = useStyles();
  const match = useRouteMatch();
  return (
    <div className={classes.root}>
      <AppBar position="fixed" className={classNames(classes.appBar)}>
        <Toolbar>
          <Grid container justify="space-between" alignItems="center" className={classes.container}>
            {!isMobile && (
              <Grid item sm="auto">
                <Link to={R.APP} className={classes.link}>
                  <Typography variant="h6">Countries</Typography>
                </Link>
              </Grid>
            )}
            {match.isExact && _isEqual(match.url, R.APP) && (
              <Grid item sm="auto">
                <SearchInput handleSearch={handleSearch} filter={filter} />
              </Grid>
            )}
          </Grid>
        </Toolbar>
      </AppBar>
      <main className={classNames(classes.content)}>{children}</main>
    </div>
  );
};
