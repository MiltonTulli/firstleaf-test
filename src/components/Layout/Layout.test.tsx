import React from 'react';
import { shallow } from 'enzyme';
import { Layout } from './Layout';

describe('<Layout>', () => {
  let props;

  beforeEach(() => {
    props = {
      handleSearch: jest.fn((country: string) => {
        return country;
      }),
      filter: false
    };
  });

  it('Should match snapshot', () => {
    const wrapper = shallow(<Layout {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render chilren', () => {
    const wrapper = shallow(
      <Layout {...props}>
        <h1 data-testid="child">test</h1>
      </Layout>
    );
    const node = wrapper.find('[data-testid="child"]');
    expect(node.exists()).toBeTruthy();
    expect(node.text()).toEqual('test');
  });
});
