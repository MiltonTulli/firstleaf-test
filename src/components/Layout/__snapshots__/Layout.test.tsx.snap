// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout> Should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-2"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        className="makeStyles-container-3"
        container={true}
        justify="space-between"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm="auto"
        >
          <Link
            className="makeStyles-link-8"
            to="/app"
          >
            <WithStyles(ForwardRef(Typography))
              variant="h6"
            >
              Countries
            </WithStyles(ForwardRef(Typography))>
          </Link>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm="auto"
        >
          <SearchInput
            filter={false}
            handleSearch={[MockFunction]}
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <main
    className="makeStyles-content-6"
  />
</div>
`;
